pallets/loans/src/types/mod.rs

```diff
@@ -101,11 +109,35 @@ impl Maturity {
 	}
 }
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+pub enum CalendarEvent {
+	/// At the end of the period, e.g. the last day of the month for a monthly
+	/// period
+	End,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+pub enum ReferenceDate {
+	/// Payments are expected every period based on an event, e.g. the end of
+	/// the period E.g. if the period is monthly and the origination date is Mar
+	/// 3, the first interest payment is expected on Mar 31.
+	CalendarDate { event: CalendarEvent },
+
+	/// Payments are expected every period relative to the origination date.
+	/// E.g. if the period is monthly and the origination date is Mar 3, the
+	/// first interest payment is expected on Apr 3.
+	OriginationDate,
+}
+
 /// Interest payment periods
 #[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
 pub enum InterestPayments {
 	/// All interest is expected to be paid at the maturity date
 	None,
+	/// Interest is expected to be paid monthly
+	Monthly(ReferenceDate),
+	/// Interest is expected to be paid twice per year
+	SemiAnnually(ReferenceDate),
 }
```

Comment:
Like too much all this part, I am wondering if it has grown enough to be placed into its own module under `types`

Reply:
Probably!